py/dml/codegen.py

```diff
@@ -1394,6 +1394,21 @@ def stmt_compound(stmt, location, scope):
     [stmt_asts] = stmt.args
     lscope = Symtab(scope)
     statements = codegen_statements(stmt_asts, location, lscope)
+    if (logging.show_porting and stmt.site.dml_version() == (1, 2)
```

Comment:
Perhaps cleaner to rewrite it to operate on AST objects instead, i.e. before calling `codegen_statements`? That way we don't need to know how `return;` is transformed into `Goto`.

Reply:
It can't be done in the parser, since it relies on `location.method().outp`. We could add some parser state that bookkeeps "current method", but I think that would be unnecessarily intrusive. There is a value in keeping porting hacks local and contained.

There is maybe a risk that we miss porting some corners; we can fix that if it appears to be a problem in practice.